src/commands/test/test_collector.py

```diff
@@ -1,93 +1,131 @@
+        for test_suite in test_suite_paths:
+            test_suites.append(self._build_test_suite(test_suite, target_function))
+
+        non_empty_test_suites = list(
+            filter(lambda test_file: (test_file.test_functions) != [], test_suites)
+        )
+        return TestCollector.Result(
+            test_suites=non_empty_test_suites,
+            test_cases_count=sum(
+                [len(test_suite.test_functions) for test_suite in non_empty_test_suites]
+            ),
+        )
 
-    def build_test_subject(self, file_path: Path):
+    @classmethod
+    def is_test_suite(cls, filename: str) -> bool:
+        return any(
+            test_re.match(filename)
+            for test_re in cls.supported_test_suite_filename_patterns
+        )
+
+    def _build_test_suite(
+        self, file_path: Path, target_function: Optional[str]
+    ) -> TestSuite:
         test_functions = self._collect_test_functions(file_path)
-        if self.target_function:
-            test_functions = [
-                f for f in test_functions if f["name"] == self.target_function
-            ]
-        return TestSubject(
+        if target_function:
+            test_functions = [f for f in test_functions if f["name"] == target_function]
+        return TestSuite(
             test_path=file_path,
             test_functions=test_functions,
         )
 
-    def get_test_files(self) -> Generator[Path, None, None]:
-        if not self.target.is_dir():
-            yield self.target
+    def _get_test_suite_paths(self, target: Path) -> Generator[Path, None, None]:
+        if not target.is_dir():
+            yield target
             return
-        for root, _, files in os.walk(self.target):
-            test_files = [Path(root, file) for file in files]
-            test_files = filter(lambda file: self.is_test_file(file.name), test_files)
-            for test_file in test_files:
-                yield test_file
-
-    @classmethod
-    def is_test_file(cls, filename: str) -> bool:
-        return any(test_re.match(filename) for test_re in cls.test_filename_re)
+        for root, _, files in os.walk(target):
+            test_suite_paths = [Path(root, file) for file in files]
+            test_suite_paths = filter(
+                lambda file: self.is_test_suite(file.name), test_suite_paths
+            )
+            for test_suite_path in test_suite_paths:
+                yield test_suite_path
 
-    def _collect_test_functions(self, file_path: Path) -> List[dict]:
+    def _collect_test_functions(
+        self, file_path: Path
+    ) -> List[StarknetCompiler.AbiElement]:
         try:
-            preprocessed = StarknetCompiler(
-                include_paths=self.include_paths or [],
-                disable_hint_validation=True,
-            ).preprocess_contract(file_path)
+            return self._starknet_compiler.get_functions(file_path, prefix="test_")
         except PreprocessorError as p_err:
             print(p_err)
```

Comment:
Why do we print it 🤔 here, we probably shouldn't as we raise it again

Reply:
`PreprocessorError` contains the line of code that caused this error. This information is hard to extract.